### PR TITLE
ZeroNonDec Hint Implementation

### PR DESCRIPTION
**Pull request by Sh0g0-1758**
Fixes: #465

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_signature.go [hunk 1]
```diff
@@ -386,3 +387,43 @@ func createDivModNPackedDivmodV1Hinter(resolver hintReferenceResolver) (hinter.H
 
 	return newDivModNPackedDivmodV1Hint(a, b), nil
 }
+
+// IsZeroNondet hint computes whether a value is zero or not
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_signature.go:391]: can you look at how comments are formatted for others hints and do the same?

File: pkg/hintrunner/zero/zerohint_signature_test.go [hunk 2]
```diff
@@ -248,5 +248,46 @@ func TestVerifyZeroHint(t *testing.T) {
...
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsZeroNondetHint()
+				},
+				check: apValueEquals(feltUint64(0)),
+			},
+			{
+				operanders: []*hintOperander{},
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					err := ctx.ScopeManager.AssignVariable("x", bigIntString("-42", 10))
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_signature_test.go:281]: `x` will never be a negative value as it is set to : 

```
x = pack(ids.x, PRIME) % SECP_P
```

**Review by TAdev0 — CHANGES_REQUESTED** [review]

**Review by TAdev0 — CHANGES_REQUESTED** [review]
I think this should not go to signature, as it just check whether a field is equal to 0 or not, module secp256k1 prime

this should go to EC hints (hints related to secp256k1 so it makes sense) i think

File: pkg/hintrunner/zero/zerohint_signature.go (hunk 1 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_signature.go]: done.

File: pkg/hintrunner/zero/zerohint_signature_test.go (hunk 2 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_signature_test.go]: done.

**Comment by Sh0g0-1758** [comment]
Right Relocated the hints.

File: pkg/hintrunner/zero/zerohint_ec_test.go [hunk 3]
```diff
@@ -846,6 +846,47 @@ func TestZeroHintEc(t *testing.T) {
...
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsZeroNondetHint()
+				},
+				check: apValueEquals(feltUint64(0)),
+			},
+		},
```
Comment (har777) [inline pkg/hintrunner/zero/zerohint_ec_test.go:889]: isn't this test redundant?

**Review by har777 — APPROVED** [review]

**Review by TAdev0 — APPROVED** [review]

File: pkg/hintrunner/zero/zerohint_ec_test.go (hunk 3 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_ec_test.go:889]: agree just one test is enough

the function takes as input a field element so it doesnt make sense to test it with negative values. currently the 3 unit tests test the same thing so just one should be enough

Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_ec_test.go]: Okay. Removing the test. But 2 tests are required, why do you say one @TAdev0  ?

File: pkg/hintrunner/zero/zerohint.go [hunk 4]
```diff
@@ -122,8 +122,8 @@ func GetHintFromCode(program *zero.ZeroProgram, rawHint zero.Hint, hintPC uint64
 		return createVerifyZeroHinter(resolver)
 	case divModNPackedDivmodV1Code:
 		return createDivModNPackedDivmodV1Hinter(resolver)
-	case isZeroPackCode:
-		return createIsZeroPackHinter(resolver)
+	case isZeroNondetCode:
+		return createIsZeroNondetHinter()
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint.go:126]: there is an issue here, you dont call the right create function

**Review by TAdev0 — CHANGES_REQUESTED** [review]

**Comment by Sh0g0-1758** [comment]
Hmm... There are some serious branch issues going on here. It seems like two branches have merged for some reason. Wait Let me correct it.

File: pkg/hintrunner/zero/zerohint.go (hunk 4 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint.go]: Right, Fixed it.

**Comment by Sh0g0-1758** [comment]
@TAdev0 , we good to merge now ?

**Review by TAdev0 — APPROVED** [review]

**Pull request merged**